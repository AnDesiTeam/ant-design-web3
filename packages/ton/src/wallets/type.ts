import type { Wallet } from '@ant-design/web3-common';
import { type WalletInfo } from '@tonconnect/sdk';

export type TonBasicWallet = {
  appName: string;
};

export type WalletWithBridge = Wallet & WalletInfo;

export type TonWalletMetadata = WalletInfo & {
  universalUrl: string;
  bridgeUrl: string;
};

export interface WalletFactory {
  create: () => WalletWithBridge;
}
