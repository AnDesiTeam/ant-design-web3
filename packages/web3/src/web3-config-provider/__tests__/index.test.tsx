import {
  Web3ConfigProvider,
  type Account,
  ConnectButton,
  useAccount,
  useNFT,
} from '@ant-design/web3';
import React from 'react';
import { render } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import type { Web3ConfigProviderProps } from '@ant-design/web3-common';
import { Spin } from 'antd';

const baseProps: Web3ConfigProviderProps = {
  availableWallets: [
    {
      name: 'MetaMask',
      remark: 'Easy-to-use browser extension.',
      extensions: [
        {
          key: 'Chrome',
          browserIcon:
            'https://github.com/ant-design/ant-design/assets/10286961/0d4e4ac7-8f89-4147-a06a-de72c02e85cb',
          browserName: 'Chrome',
          link: 'https://chrome.google.com/webstore/detail/metamask/nkbihfbeogaeaoehlefnkodbefgpgknn',
          description: 'Access your wallet right from your favorite web browser.',
        },
      ],
    },
  ],
};

describe('web3-config-provider', () => {
  it('Should work', () => {
    const App: React.FC = () => {
      const [account, setAccount] = React.useState<Account>();
      return (
        <Web3ConfigProvider
          {...baseProps}
          connect={async () => {
            const newAccount = {
              address: '0x1234567890123456789012345678901234567890',
            };
            setAccount(newAccount);
          }}
          account={account}
        >
          <ConnectButton />
        </Web3ConfigProvider>
      );
    };

    const { baseElement } = render(<App />);
    expect(baseElement).toMatchSnapshot();
  });
  it('useAccount', () => {
    const Child: React.FC = () => {
      const { account } = useAccount();
      return <div>{account?.address}</div>;
    };
    const App: React.FC = () => {
      const [account, setAccount] = React.useState<Account>();
      return (
        <Web3ConfigProvider
          {...baseProps}
          connect={async () => {
            const newAccount = {
              address: '0x1234567890123456789012345678901234567890',
            };
            setAccount(newAccount);
          }}
          account={account}
        >
          <Child />
        </Web3ConfigProvider>
      );
    };

    const { baseElement } = render(<App />);
    expect(baseElement).toMatchSnapshot();
  });
  it('useNFT', () => {
    const Child: React.FC = () => {
      const { metadata, error, loading } = useNFT(
        '0x79fcdef22feed20eddacbb2587640e45491b757f',
        42n,
      );
      if (error) {
        return <div>{error.message}</div>;
      }
      return <Spin spinning={loading}>{metadata.name}</Spin>;
    };
    const App: React.FC = () => {
      const [account, setAccount] = React.useState<Account>();
      return (
        <Web3ConfigProvider
          {...baseProps}
          connect={async () => {
            const newAccount = {
              address: '0x1234567890123456789012345678901234567890' as `0x${string}`,
            };
            setAccount(newAccount);
          }}
          account={account}
        >
          <Child />
        </Web3ConfigProvider>
      );
    };

    const { baseElement } = render(<App />);
    expect(baseElement).toMatchSnapshot();
  });
  it('useNFT with custom getNFTMetadata', () => {
    const Child: React.FC = () => {
      const { metadata, error, loading } = useNFT(
        '0x79fcdef22feed20eddacbb2587640e45491b757f',
        42n,
      );
      if (error) {
        return <div>{error.message}</div>;
      }
      return <Spin spinning={loading}>{metadata.name}</Spin>;
    };
    const App: React.FC = () => {
      const [account, setAccount] = React.useState<Account>();
      return (
        <Web3ConfigProvider
          {...baseProps}
          connect={async () => {
            const newAccount = {
              address: '0x1234567890123456789012345678901234567890' as `0x${string}`,
            };
            setAccount(newAccount);
          }}
          account={account}
          getNFTMetadata={() => {
            return Promise.resolve({
              attributes: [],
              name: 'test',
              date: Date.now(),
              description: 'test',
              image: 'https://no.com/z.png',
            });
          }}
        >
          <Child />
        </Web3ConfigProvider>
      );
    };

    const { baseElement } = render(<App />);
    expect(baseElement).toMatchSnapshot();
  });
});
