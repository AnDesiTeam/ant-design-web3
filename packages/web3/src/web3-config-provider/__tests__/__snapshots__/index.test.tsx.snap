// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`web3-config-provider > Should work 1`] = `
<body>
  <div>
    <button
      class="ant-btn css-dev-only-do-not-override-1ck3jst ant-btn-default css-dev-only-do-not-override-1ck3jst ant-web3-connect-button"
      type="button"
    >
      <div
        class="ant-web3-connect-button-text css-dev-only-do-not-override-1ck3jst"
      >
        Connect Wallet
      </div>
    </button>
  </div>
</body>
`;

exports[`web3-config-provider > useAccount 1`] = `
<body>
  <div>
    <div />
  </div>
</body>
`;

exports[`web3-config-provider > useNFT 1`] = `
<body>
  <div>
    <div>
      Provider is not ready
    </div>
  </div>
</body>
`;

exports[`web3-config-provider > useNFT with custom getNFTMetadata 1`] = `
<body>
  <div>
    <div
      aria-busy="false"
      aria-live="polite"
      class="ant-spin css-dev-only-do-not-override-1ck3jst"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </div>
  </div>
</body>
`;
